### ISSUE TITLE
Confusing documentation for default segment args

### ISSUE THREAD

**Issue by brlodi** [powerline/powerline#1820]
After spending quite a bit of time diagnosing why a `hostname` segment in a shell theme wouldn't appear, including carefully checking the [documentation](https://powerline.readthedocs.io/en/master/configuration/segments/common.html#powerline.segments.common.net.hostname) and the [source it links to](https://powerline.readthedocs.io/en/master/_modules/powerline/segments/common/net.html#hostname) (and [on GitHub](/powerline/powerline/blob/develop/powerline/segments/common/net.py#L17)), I finally realized that the `only_in_ssh` argument is set to true in `.../themes/shell/__main__.json`. I had assumed, I think fairly, that optional segment arguments wouldn't be quietly overridden by the default config files.

It would be great if the docs could include a hint that, despite the declaration setting the parameter to `false` by default in the source (and therefore the docs), it is overridden by default for an entire environment. I think I disagree with that override in general, but at the very least I think it would be very helpful to document the full set of defaults for segments in the out-of-the-box config.

**Comment by ZyX-I** [comment]
It is overriden by default for shell only, other bindings have that `False` by default.

**Comment by brlodi** [comment]
Yes I see that, and I think having that as an undocumented thing is an unnecessary source of confusion.
